src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,808 @@
+   end
+end
+
+function Base.show(io::IO, x::MatrixGroupElem)          #TODO: is this correct?
+   if isdefined(x, :elm)
+      display(x.elm)
+   else
+      print(io, GAP.gap_to_julia(GAP.Globals.StringViewObj(x.X)))
+   end
+end
+
+group_element(G::MatrixGroup, x::GapObj) = MatrixGroupElem(G,x)
+
+function Base.getproperty(G::MatrixGroup, sym::Symbol)
+
+   if sym === :mat_iso || sym === :ring_iso
+      if isdefined(G,sym) return getfield(G,sym) end
+      fm = gen_mat_iso(G.deg, G.ring)
+      setfield!(G,:ring_iso,fm.fr)
+      setfield!(G,:mat_iso,fm)
+
+   elseif sym === :X
+      if isdefined(G,:X) return getfield(G,:X) end
+      if isdefined(G,:descr)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            setfield!(G,:ring_iso,fm.fr)
+            setfield!(G,:mat_iso,fm)
+         end
+         if G.descr==:GL setfield!(G,:X,GAP.Globals.GL(G.deg,G.ring_iso.codomain))
+         elseif G.descr==:SL setfield!(G,:X,GAP.Globals.SL(G.deg,G.ring_iso.codomain))
+         elseif G.descr==:Sp setfield!(G,:X,GAP.Globals.Sp(G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("GO+") setfield!(G,:X,GAP.Globals.GO(1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("SO+") setfield!(G,:X,GAP.Globals.SO(1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("Omega+") setfield!(G,:X,GAP.Globals.Omega(1,G.deg,Int(order(G.ring))))
+         elseif G.descr==Symbol("GO-") setfield!(G,:X,GAP.Globals.GO(-1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("SO-") setfield!(G,:X,GAP.Globals.SO(-1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("Omega-") setfield!(G,:X,GAP.Globals.Omega(-1,G.deg,Int(order(G.ring))))
+         elseif G.descr==:GO setfield!(G,:X,GAP.Globals.GO(0,G.deg,G.ring_iso.codomain))
+         elseif G.descr==:SO setfield!(G,:X,GAP.Globals.SO(0,G.deg,G.ring_iso.codomain))
+         elseif G.descr==:Omega setfield!(G,:X,GAP.Globals.Omega(0,G.deg,Int(order(G.ring))))
+         elseif G.descr==:GU setfield!(G,:X,GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) )))
+         elseif G.descr==:SU setfield!(G,:X,GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) )))
+         end
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            setfield!(G,:ring_iso,fm.fr)
+            setfield!(G,:mat_iso,fm)
+         end
+         V = GAP.julia_to_gap([g.X for g in G.gens])
+         setfield!(G,:X,GAP.Globals.Group(V))
+      end
+
+   elseif sym === :gens
+      if isdefined(G, :gens) return getfield(G, :gens) end
+      if !isdefined(G, :X)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            setfield!(G,:ring_iso,fm.fr)
+            setfield!(G,:mat_iso,fm)
+         end
+         if G.descr==:GL setfield!(G,:X,GAP.Globals.GL(G.deg,G.ring_iso.codomain))
```

Comment:
It would be great to not duplicate this long chain of `if-elsefi` here, and for example to factor it out in a function.